### PR TITLE
feat: generate image bom for Windows VHDs

### PR DESCRIPTION
**Pull request by AbelHu**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
generate image bom for Windows VHDs

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: vhdbuilder/packer/list-images.ps1 [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        # GiB
+        # windows/amd64
+        # io.cri-containerd.image=managed
+        $repoTags=$splitResult[0]
+        $id=$splitResult[2]
+
+        $isExist=$false
+        foreach($bom in $bomList) {
+            if ($bom.id -eq $id) {
+                $bom.repoTags += $repoTags
```
Comment (alexeldeib) [inline vhdbuilder/packer/list-images.ps1:68]: we don't add this for linux I think. we only have the repo digest? I believe I checked with Gaurav and he didn't need the image id (hash of layers), only the manifest sha (which you have already).

may want to double check with Gaurav

**Review by alexeldeib — APPROVED** [review]
lgtm.  two minor comment you may want to check with Gaurav/FIM folks -- 

Linux has very few cases where the same image has multiple tags, so the sha/tag are mostly 1:1. We have one case and it hasn't caused issues, but if _most_ windows images are tagged twice FIM folks may need to handle it in their pipelines.

other comment in code

File: vhdbuilder/packer/list-images.ps1 (hunk 1 above)
Comment (alexeldeib) [reply vhdbuilder/packer/list-images.ps1:68]: as ref https://github.com/Azure/AgentBaker/blob/7a0ef4ff0090e9581ccfc97e5c3b4477db1863ad/vhdbuilder/release-notes/AKSUbuntu/gen2/2204containerd/latest-image-list.json#L509-L518

Comment (alexeldeib) [reply vhdbuilder/packer/list-images.ps1:68]: sorry -- not add it as part of repoTags. it's already there as `id`.

**Comment by AbelHu** [comment]
> lgtm. two minor comment you may want to check with Gaurav/FIM folks --
> 
> Linux has very few cases where the same image has multiple tags, so the sha/tag are mostly 1:1. We have one case and it hasn't caused issues, but if _most_ windows images are tagged twice it may need to be handled in ingestion pipelines.
> 
> other comment in code

Thanks for the feedback, @alexeldeib. I think that we can remove sha256tag as you said since the sha256tag is id.

Is retagMCRImagesForChina not used by Linux VHDs? It seems like that it is needed for AzureChinaCloud. Please free to correct me.

File: vhdbuilder/packer/list-images.ps1 (hunk 1 above)
Comment (AbelHu) [reply vhdbuilder/packer/list-images.ps1]: Thanks for the feedback. Add the code to ignore repoTags when it contains id.

**Pull request merged**